includes/sanitizers/class-amp-audio-sanitizer.php

```diff
@@ -31,14 +31,28 @@ public function sanitize() {
 
 			$new_node = AMP_DOM_Utils::create_node( $this->dom, 'amp-audio', $new_attributes );
 
-			// TODO: limit child nodes too (only allowed: `source`; move rest to div+fallback)
 			// TODO: `source` does not have closing tag, and DOMDocument doesn't handle it well.
 			foreach ( $node->childNodes as $child_node ) {
 				$new_child_node = $child_node->cloneNode( true );
-				$new_node->appendChild( $new_child_node );
+				$old_child_attributes = AMP_DOM_Utils::get_node_attributes_as_assoc_array( $new_child_node );
+				$new_child_attributes = $this->filter_attributes( $old_child_attributes );
```

Comment:
Not going to block merging (I will open a new issue for this) but we should probably filter attributes for the `source` elements separately (since the general attributes don't apply for it).